docs/sources/developers/plugins/create-a-grafana-plugin/extend-a-plugin/add-log-queries.md

```diff
@@ -0,0 +1,508 @@
+
+```ts
+import {
+  DataQueryRequest,
+  DataQueryResponse,
+  DataSourceWithLogsContextSupport,
+  LogRowContextOptions,
+  LogRowContextQueryDirection,
+  LogRowModel,
+} from '@grafana/data';
+import { catchError, lastValueFrom, of, switchMap, Observable } from 'rxjs';
+
+export class ExampleDatasource implements DataSourceWithLogsContextSupport<ExampleQuery> {
+  // Retrieve context for a given log row
+  async getLogRowContext(
+    row: LogRowModel,
+    options?: LogRowContextOptions,
+    query?: ExampleQuery
+  ): Promise<DataQueryResponse> {
+    // createRequestFromQuery is a mocked implementation. Be sure to adjust this based on your data source logic.
+    const request = createRequestFromQuery(row, query, options);
+    return lastValueFrom(
+      // this.query is a mocked implementation. Be sure to adjust this based on your data source logic.
+      this.query(request).pipe(
+        catchError((err) => {
+          const error: DataQueryError = {
+            message: 'Error during context query. Please check JS console logs.',
+            status: err.status,
+            statusText: err.statusText,
+          };
+          throw error;
+        }),
+        // processResultsToDataQueryResponse is a mocked implementation. Be sure to adjust this based on your data source logic.
+        switchMap((res) => of(processResultsToDataQueryResponse(res)))
+      )
+    );
+  }
+
+  // Retrieve the context query object for a given log row. This is currently used to open LogContext queries in a split view.
+  getLogRowContextQuery(
+    row: LogRowModel,
+    options?: LogRowContextOptions,
+    query?: ExampleQuery
+  ): Promise<ExampleQuery | null> {
+    // Data source internal implementation that creates context query based on row, options and original query
+  }
+
+  // This method can be used to show "context" button based on runtime conditions (for example row model data or plugin settings,...)
+  showContextToggle(row?: LogRowModel): boolean {
+    // If you want to always show toggle, you can just return true
+    if (row && row.searchWords && row.searchWords.length > 0) {
+      return true;
+    }
+  }
+}
+```
+
+### Filtering of fields using Log details
+
+> To be implemented trough data source method
+
+Every log line has an expandable part called "Log details" that can be opened by clicking on the line. Within Log details, [Fields]({{< relref "../../../../explore/logs-integration/#fields" >}}) associated with that log entry are displayed. If the data source implements `modifyQuery?(query: TQuery, action: QueryFixAction): TQuery;` API, then filtering functionality is available for each field. For logs, two filtering options are currently available:
+
+- `ADD_FILTER` - to filter for log lines that include selected fields
+- `ADD_FILTER_OUT` - to filter for log lines that do not include selected fields
+
+```ts
+export class ExampleDatasource implements DataSourceWithSupplementaryQueriesSupport<ExampleQuery> {
```

Comment:
I think `DataSourceWithSupplementaryQueriesSupport<ExampleQuery>` is not needed in this example.

Reply:
Yes, I'll change that, good catch. Copy-paste issue.